[AriellaCallista] iP

AriellaCallista opened the pull request:
### SanaPro

> "Chaos is in the world. Not in your head" - David Allen [(source)](https://medium.com/the-mission/the-most-popular-productivity-pieces-of-wisdom-from-david-allen-72ffe70ac7b9#:~:text=%E2%80%9CChaos%20is%20in%20the%20world.%20Not%20in%20your%20head.%E2%80%9D)

SanaPro helps keep the chaos of the world from entering your head by allowing you to keep track of your todos. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://github.com/AriellaCallista/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💯 

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Event reminders

___

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Add greeting and exit

Commit:
Improve skeletal version to echo message and exit

Commit:
Implement add and list functionalities

Commit:
Create class for task

Commit:
Improove efficiency by using StringBuilder instead of String in loop

Commit:
Implement mark as done functionality

AriellaCallista committed:
Replace getDescription with toString method

AriellaCallista committed:
Create subclasses for tasks

AriellaCallista committed:
Replace loop with scanner methods and implement todo

AriellaCallista committed:
Implement adding deadline type task

AriellaCallista committed:
Implement adding event type task

Commit:
Create DukeException class

Commit:
Implement error handling

Commit:
Implement deleting task

Commit:
Implement enums for commands

Commit:
Implement automated text ui testing

Commit:
Refactor class name to Sana

Commit:
Add more test cases

Commit:
Make helper methods in Sana class private

Commit:
Implement save tasks function

Commit:
Merge branch 'branch-level-7'

Commit:
Implement date and time for deadline

Commit:
Implement date and time for event

Commit:
Merge branch 'branch-level-8'

Commit:
Create classes to make code more OOP

Commit:
Implement parser class to parse commands

Commit:
Implement UI class

Commit:
Implement TaskList class

Commit:
Implement load and save functions in Storage class

Commit:
Implement add todo task function

Commit:
Implement add deadline task function

Commit:
Bugfix for empty todo command

Commit:
Bugfix for doubled lines on printing error msg

Commit:
Implement Command superclass

Commit:
Implement adding event task function

Commit:
Implement HelpCommand for invalid command words

Commit:
Implement ExitCommand class

Commit:
Implement listing tasks function

Commit:
Implement formatString method for saving tasks

Commit:
Task: Implement getTask

Commit:
AddCommand: refactor constructor to include isDone

Commit:
Refactor save method to write formatted tasks

Commit:
Refactor tasksList to be an array list of tasks

Commit:
ListCommand: implement execute method

Commit:
Refactor deleting tasks function

Commit:
Refactor unmark and mark tasks functions

Commit:
Merge branch 'branch-more-OOP'

Commit:
Package classes into 'sana'

Commit:
Merge branch 'add-gradle-support'

Commit:
build.gradle: set mainClass to sana.Sana

Commit:
Update files for text ui test

Commit:
Ui: refactor class to have scanner as field

Commit:
Add JUnit tests for Task class formatTask method

Commit:
Add JUnit tests for parser method

Commit:
Add JUnit tests for AddCommand execute method

Commit:
Add header comments to non-private classes/methods

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Add checkstyle configuration

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Implement find task function

Commit:
Merge branch 'branch-Level-9'

Commit:
Update runtest.sh so it runs without existing data

ss-shah commented on src/main/java/sana/Sana.java:
```diff
@@ -0,0 +1,49 @@
+            try {
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                Command c = Parser.parse(fullCommand);
```
```java
Command c = Parser.parse(fullCommand);
```
Variable `c` should have a more descriptive name, such as `parsedCommand`.

ss-shah commented on src/main/java/sana/Storage.java:
```diff
@@ -0,0 +1,87 @@
+        File folder = new File(folderPath);
+
+        // Check if folder exists, if not create one
+        if (!folder.exists()) {
```
`if (!folder.exists()) {`

May alternatively be handled by catching the `FileNotFoundException`.

ss-shah commented on src/main/java/sana/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+     *
+     * @return the number of tasks in the list.
+     */
+    public int size() {
```
`public int size() {`

Method names should be verbs, eg. `getSize()`.

ss-shah commented on src/main/java/sana/AddCommand.java:
```diff
@@ -0,0 +1,136 @@
+                storage.save("/Users/ariellacallista/Desktop",
+                        "/Users/ariellacallista/Desktop/SanaTasks.txt", newTodo);
+
+                System.out.println("Got it. I've added this task:\n" + newTodo + "\n"
```
`System.out.println("Got it. I've added this task:\n" + newTodo + "\n"`

Based on your class structure, all of the outputs (eg. `system.out.println()`) should be in the Ui class to make it easier to change to a GUI. You can instead add a `showTaskAddedMessage()` method to the Ui class and call it here. Similar for all of the other types of tasks.

ss-shah commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,24 @@
+event read book /from 2023-08-30 /to 2023-08-31
+event read book /from 2023-08-30 /to 2023-08-31
+invalid command
+bye
```
This file should probably be untracked (added to .gitignore)

LordSaumya reviewed (commented):
Well designed with a clear class hierarchy, mostly in adherence to coding and Javadoc standards. Few minor changes required.

Commit:
Add GUI using JavaFX

Commit:
Merge branch 'branch-Level-10'

Commit:
Add javadocs and fix code standard violations

Commit:
Refactor filepath in storage

Commit:
Create new jar file

Commit:
Add assertions

Commit:
Refactor code to adhere to code quality

Commit:
AddCommand: replace class with the smaller classes

The execute method in the AddCommand class is getting too long
(over 100 LoC)

Separating the execute method in different classes is more readable and
maintainable in the long run as opposed to using one large switch
statement in the AddCommand class.

Let's extract the classes based on each command i.e. AddTodoClass,
AddDeadlineClass, AddEventClass.

Commit:
Merge branch 'branch-A-CodeQuality' of https://github.com/AriellaCallista/ip into branch-A-CodeQuality

AriellaCallista committed:
Merge pull request #1 from AriellaCallista/branch-A-Assertions

Add assertions

Commit:
Merge branch 'master' of https://github.com/AriellaCallista/ip

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	data/SanaTasks.txt
#	src/main/java/sana/AddCommand.java
#	src/main/java/sana/DeleteCommand.java
#	src/main/java/sana/Parser.java
#	src/main/java/sana/TaskList.java

AriellaCallista committed:
Merge pull request #2 from AriellaCallista/branch-A-CodeQuality

Commit:
Merge branch 'master' of https://github.com/AriellaCallista/ip

AriellaCallista committed:
Create gradle.yml

AriellaCallista committed:
Merge pull request #3 from AriellaCallista/AriellaCallista-patch-1

Create gradle.yml

Commit:
Merge branch 'master' of https://github.com/AriellaCallista/ip

Commit:
Task class: add update 'this' task function

Commit:
Parser parse: add update case

Commit:
TaskList class: add update task list function

Commit:
UpdateCommand class: add class for updating tasks.

Need to implement an update task command.

Extracting a class for the update command allows us to use polymorphism
when dealing with command instances.

Let's implement an execute method for UpdateCommand class which updates
the task at a particular given index in the task list.

AriellaCallista committed:
Merge pull request #4 from AriellaCallista/branch-C-Update

Branch c update

Commit:
Merge branch 'master' of https://github.com/AriellaCallista/ip

Commit:
Set up product website

Commit:
Remove println statement

Commit:
Fix bugs

Commit:
Fix code quality

Commit:
Final version of ip

Commit:
Remove import platform.exit()

Commit:
Add scan.close()

Commit:
Fix code quality
